samples/quadratic-assignment-problem/quadratic-assignment-problem.ipynb

```diff
@@ -0,0 +1,393 @@
+    "\n",
+    "$$  P \\cdot (\\sum_{i} c_{i} \\cdot x_{i} + k)^2 \\quad $$\n",
+    "\n",
+    "In our example, $k = -1$ and $c_i = 1$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of monimial terms: 36, number of grouped terms: 6\n",
+      "Grouped terms: [{'c': 1000, 'terms': [{'c': 1, 'ids': [0]}, {'c': 1, 'ids': [1]}, {'c': 1, 'ids': [2]}, {'c': -1, 'ids': []}]}, {'c': 1000, 'terms': [{'c': 1, 'ids': [3]}, {'c': 1, 'ids': [4]}, {'c': 1, 'ids': [5]}, {'c': -1, 'ids': []}]}, {'c': 1000, 'terms': [{'c': 1, 'ids': [6]}, {'c': 1, 'ids': [7]}, {'c': 1, 'ids': [8]}, {'c': -1, 'ids': []}]}, {'c': 1000, 'terms': [{'c': 1, 'ids': [0]}, {'c': 1, 'ids': [3]}, {'c': 1, 'ids': [6]}, {'c': -1, 'ids': []}]}, {'c': 1000, 'terms': [{'c': 1, 'ids': [1]}, {'c': 1, 'ids': [4]}, {'c': 1, 'ids': [7]}, {'c': -1, 'ids': []}]}, {'c': 1000, 'terms': [{'c': 1, 'ids': [2]}, {'c': 1, 'ids': [5]}, {'c': 1, 'ids': [8]}, {'c': -1, 'ids': []}]}]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# here we set penalty to the largest coefficient in the original objective function (although we can go higher than this)\n",
+    "P = 1000\n",
+    "k = -1\n",
+    "\n",
+    "# construct facility constraints\n",
+    "for i in range(n):    \n",
+    "    terms_list.append(SlcTerm(\n",
+    "        c = P,\n",
+    "        terms = [\n",
+    "            Term(\n",
+    "                c = 1,\n",
+    "                indices = [variable_map[(i,j)]]\n",
+    "            )\n",
+    "        for j in range(n)\n",
+    "        ] + \n",
+    "        [Term(c=k, indices=[])] #constant k term\n",
+    "    ))\n",
+    "\n",
+    "# construct location constraints\n",
+    "for i in range(n):    \n",
+    "    terms_list.append(SlcTerm(\n",
+    "        c = P,\n",
+    "        terms = [\n",
+    "            Term(\n",
+    "                c = 1,\n",
+    "                indices = [variable_map[(j,i)]]\n",
+    "            )\n",
+    "        for j in range(n)\n",
+    "        ] + \n",
+    "        [Term(c=k, indices=[])] #constant k term\n",
+    "    ))\n",
+    "\n",
+    "# construct final problem with all the terms (monomial and slc)\n",
+    "problem = Problem(name=\"Small QAP\", problem_type=ProblemType.pubo, terms=terms_list)\n",
+    "\n",
+    "print(f\"Number of monimial terms: {len(problem.terms)}, number of grouped terms: {len(problem.terms_slc)}\")\n",
+    "print(f\"Grouped terms: {problem.terms_slc}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Submitting problem to the Azure Quantum solver\n",
+    "\n",
+    "The SlcTerm functionality is only available on the population annealing and substochastic monte carlo solvers. For this sample we will use the population annealing solver with its default parameters (since the problem is trivial).\n",
```

Comment:
```suggestion
    "The SlcTerm functionality is only currently available on the population annealing and substochastic monte carlo solvers. For this sample we will use the population annealing solver with its default parameters (since the problem is trivial).\n",
```

Unless we aren't planning to enable SLC for other solvers in future, in which case that's fine :)

Reply:
Would capitalise the solver names and link to the docs:
```suggestion
    "The SlcTerm functionality is only available on the [Population Annealing](https://docs.microsoft.com/azure/quantum/optimization-population-annealing#parameterized-population-annealing) and [Substochastic Monte Carlo](https://docs.microsoft.com/azure/quantum/optimization-substochastic-monte-carlo#parameterized-substochastic-monte-carlo) solvers. For this sample we will use the population annealing solver with its default parameters (since the problem is trivial).\n",
```

Reply:
I think "only available" is fine as that's factual for now. It's unclear how quickly we'll move this out of early access but solver support will grow as solvers move onto our new backend.

Reply:
Because the population annealing and substochastic monte carlo algorithm are packaged in solver products we should always spell them with capitals. (also the third mention of Population Annealing that is missing from Anita's suggestion)